File: Serene/Serene.Web/Modules/Northwind/Product/ProductDialog.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    @Serenity.Decorators.responsive()
    @Serenity.Decorators.maximizable()
    export class ProductDialog extends Serenity.EntityDialog<ProductRow, any> {
        protected getFormKey() { return ProductForm.formKey; }
        protected getIdProperty() { return ProductRow.idProperty; }
        protected getLocalTextPrefix() { return ProductRow.localTextPrefix; }
        protected getNameProperty() { return ProductRow.nameProperty; }
        protected getService() { return ProductService.baseUrl; }

        protected form = new ProductForm(this.idPrefix);

        protected getLanguages(): string[][] {
            return LanguageList.get_value();
        }
    }
}

File: Serene/Serene.Script/Northwind/Product/ProductDialog.cs
namespace Serene.Northwind
{
    using Serenity;
    using System.Runtime.CompilerServices;

    [Imported]
    public class ProductDialog : EntityDialog<ProductRow>
    {
    }
}
/* 
This class has been ported to TypeScript. See ProductGrid.ts
Code below is only a reference for those who want to use Saltaralle

namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;
    using System.Collections.Generic;

    [IdProperty("ProductID"), NameProperty("ProductName")]
    [FormKey("Northwind.Product"), LocalTextPrefix("Northwind.Product"), Service("Northwind/Product")]
    public class ProductDialog : EntityDialog<ProductRow>
    {
        protected override IEnumerable<System.Tuple<string, string>> GetLanguages()
        {
            return LanguageList.Value;
        }
    }
}
*/